openstack/compute/v2/servers/requests.go

```diff
@@ -148,24 +146,14 @@ type CreateOpts struct {
 	// Name is the name to assign to the newly launched server.
 	Name string `json:"name" required:"true"`
 
-	// ImageRef [optional; required if ImageName is not provided] is the ID or
-	// full URL to the image that contains the server's OS and initial state.
+	// ImageRef is the ID or full URL to the image that contains the
+	// server's OS and initial state.
 	// Also optional if using the boot-from-volume extension.
 	ImageRef string `json:"imageRef"`
 
-	// ImageName [optional; required if ImageRef is not provided] is the name of
-	// the image that contains the server's OS and initial state.
-	// Also optional if using the boot-from-volume extension.
-	ImageName string `json:"-"`
```

Comment:
Was removing the `ImageName` and `FlavorName` options intended? It was a nice convenience to be able to pass the service client and not have to worry about calling `IDFromName(...)` ourselves. If so, it might be worthwhile to remove the `ServiceClient` as an option from the `CreateOpts` here and possibly other packages as it serves no purpose anymore.

Reply:
Good catch. Yes, it was intended though I do agree that it removes some well-received convenience.